lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -474,14 +474,21 @@ previewers.vimgrep = defaulter(function(opts)
         pcall(vim.api.nvim_buf_clear_namespace, self.state.last_set_bufnr, ns_previewer, 0, -1)
       end
 
-      conf.buffer_previewer_maker(p, self.state.bufnr, {
-        bufname = self.state.bufname,
-        winid = self.state.winid,
-        preview = opts.preview,
-        callback = function(bufnr)
-          jump_to_line(self, bufnr, entry.lnum)
-        end,
-      })
+      -- Workaround for unnamed buffer when using builtin.buffer
+      if entry.bufnr and (p == "[No Name]" or vim.api.nvim_buf_get_option(entry.bufnr, "buftype") ~= "") then
+        local lines = vim.api.nvim_buf_get_lines(entry.bufnr, 0, -1, false)
+        vim.api.nvim_buf_set_lines(self.state.bufnr, 0, -1, false, lines)
+        jump_to_line(self, self.state.bufnr, entry.lnum)
+      else
```

Comment:
I didnt run into any issue but we should also do this if else in `term` buffer (we had it there before as well).

Or we just dont set `previewer = conf.grep_previewer(opts)` but rather set `previewer = previewers.vim_buffer_vimgrep.new(opts)` and basically say you are not allowed to have bat here :laughing:

Reply:
My bad, missed it. Reverted that segment too and briefly tested it with

```lua
require "telescope.builtin".buffers { previewer = require "telescope.previewers".vimgrep.new({}) }
```

Once I installed `bat`, it worked as intended :laughing: